src/qibolab/instruments/qm.py

```diff
@@ -527,15 +531,22 @@ def play_pulses(qmsequence, relaxation_time=0):
                 clock[qmpulse.element] += 4 * wait_cycles
             clock[qmpulse.element] += qmpulse.duration
             if pulse.type.name == "READOUT":
-                measure(
-                    qmpulse.operation,
-                    qmpulse.element,
-                    None,
-                    dual_demod.full("cos", "out1", "sin", "out2", qmpulse.I),
-                    dual_demod.full("minus_sin", "out1", "cos", "out2", qmpulse.Q),
-                )
-                if qmpulse.threshold is not None:
-                    assign(qmpulse.shot, qmpulse.I * qmpulse.cos - qmpulse.Q * qmpulse.sin > qmpulse.threshold)
+                if raw_adc:
+                    measure(
+                        qmpulse.operation,
+                        qmpulse.element,
+                        qmpulse.raw_adc,
+                    )
+                else:
+                    measure(
+                        qmpulse.operation,
+                        qmpulse.element,
+                        None,
+                        dual_demod.full("cos", "out1", "sin", "out2", qmpulse.I),
+                        dual_demod.full("minus_sin", "out1", "cos", "out2", qmpulse.Q),
+                    )
+                    if qmpulse.threshold is not None:
+                        assign(qmpulse.shot, qmpulse.I * qmpulse.cos - qmpulse.Q * qmpulse.sin > qmpulse.threshold)
```

Comment:
Since it's growing, I'd suggest splitting into a `readout` function.
In principle, you'd only need `readout(qmpulse, dual_demod, raw_adc)`.

Reply:
Done.